adapters/rest-adapter-generator/src/main/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator.java

```diff
@@ -63,4 +68,53 @@ private void validate(final GeneratorConfig configuration) {
         isTrue(outputDirectory.canWrite(), format("%s can't be written to", outputDirectory));
     }
 
+    /**
+     * Write class to the base package provided in the configuration
+     *
+     * @param typeSpec      the typeSpec to write to file
+     * @param configuration the configuration that provides the base package name
+     */
+    private void writeToBasePackage(final TypeSpec typeSpec, final GeneratorConfig configuration) {
+        writeToBasePackage(Collections.singletonList(typeSpec), configuration, "");
+    }
+
+    /**
+     * Write a list of classes to the resource package.
+     *
+     * @param typeSpecs     the list of typeSpecs to write to file
+     * @param configuration the configuration that provides the base package name
+     * @return the list of class names written to file
+     */
+    private List<String> writeToResourcePackage(final List<TypeSpec> typeSpecs, final GeneratorConfig configuration) {
+        return writeToBasePackage(typeSpecs, configuration, RESOURCE_PACKAGE_NAME);
+    }
+
+    /**
+     * Write a list of classes to a specified package.
+     *
+     * @param typeSpecs     the list of typeSpecs to write to file
+     * @param configuration the configuration that provides the base package name
+     * @return the list of class names written to file
+     * @throws IllegalStateException if an IOException is thrown while writing to file
+     */
+    private List<String> writeToBasePackage(final List<TypeSpec> typeSpecs,
+                                            final GeneratorConfig configuration,
+                                            final String packageName) {
+
+        final List<String> implementationNames = new ArrayList<>();
```

Comment:
Same as above, no harm in setting to final, even if the internal representation of the object itself is updated